cro-clib/src/basic_wallet.rs

```diff
@@ -0,0 +1,185 @@
+use crate::types::{CroAddress, CroAddressPtr, CroResult};
+use chain_core::init::address::RedeemAddress;
+
+use chain_core::state::account::StakedStateAddress;
+use chain_core::tx::data::address::ExtendedAddr;
+use client_common::MultiSigAddress;
+use client_common::{PrivateKey, PublicKey};
+use std::ptr;
+
+/// # Safety
+fn do_cro_basic_create_staking_address(private: PrivateKey) -> Result<Box<CroAddress>, ()> {
```

Comment:
can this fail? why `Result`, and not the `Box` directly?

Reply:
ok